dashboard/app/models/user_level.rb

```diff
@@ -138,15 +130,29 @@ def before_unsubmit
     end
   end
 
-  def has_autolocked?(stage)
-    return false unless stage.lockable?
-    unlocked_at && unlocked_at < AUTOLOCK_PERIOD.ago
+  # `locked` is a virtual attribute because it relies on `unlocked_at` to
+  # automatically return `true` after `AUTOLOCK_PERIOD`
+  def locked
+    unlocked_at.nil? || unlocked_at < AUTOLOCK_PERIOD.ago
```

Comment:
It takes a moment to parse what `unlocked_at < AUTOLOCK_PERIOD.ago` means. For me, I think it would be helpful to create a new method called `has_been_autolocked?` which returns `unlocked_at < AUTOLOCK_PERIOD.ago` just so it's easier to understand. But I'll leave it up to you since adding a whole new method just for that is debatable.

Reply:
@cforkish and I talked about this for a while, and actually deprecated a helper method called has_autolocked because it's only used in this single context to update the value of "locked." This refactor was to make it so nobody int he future has to think about the unlocked_at field at all, and can just interact with this method 'locked' directly.